### PR TITLE
[new release] digestif (1.0.0)

### PR DESCRIPTION
**Pull request by dinosaure**
Hashes implementations (SHA*, RIPEMD160, BLAKE2* and MD5)

- Project page: <a href="https://github.com/mirage/digestif">https://github.com/mirage/digestif</a>
- Documentation: <a href="https://mirage.github.io/digestif/">https://mirage.github.io/digestif/</a>

##### CHANGES:

- **breaking changes** Upgrade the library with MirageOS 3.9 (new layout of artifacts)
  Add tests about compilation of unikernels (execution and link)
  (mirage/digestif#105, @dinosaure, @hannesm)
- Fix `esy` installation (mirage/digestif#104, @dinosaure)
- **breaking changes** Better GADT (mirage/digestif#103, @dinosaure)
  As far as I can tell, nobody really use this part of `digestif`.
  The idea is to provide a GADT which contains the type of the hash.
  From third-part libraries point-of-view, it's better to _pattern-match_ with
  such information instead to use a polymorphic variant (as before).
- **breaking changes** key used for HMAC is a constant `string` (mirage/digestif#101, @dinosaure, @hannesm)
  The key should not follow the same type as the digest value (`string`, `bytes`, `bigstring`).
  This update restricts the user to user only constant key (as a `string`).

### CODE REVIEW

**Comment by mseri** [comment]
all versions of `git` between 2.0.0 and 2.1.3 are failing with
```
#=== ERROR while compiling git.2.1.3 ==========================================#
# context              2.0.7 | linux/x86_64 | ocaml-base-compiler.4.11.1 | file:///home/opam/opam-repository
# path                 ~/.opam/4.11/.opam-switch/build/git.2.1.3
# command              ~/.opam/4.11/bin/dune build -p git -j 72
# exit-code            1
# env-file             ~/.opam/log/git-25-5bb628.env
# output-file          ~/.opam/log/git-25-5bb628.out
### output ###
#       ocamlc src/git/.git.objs/byte/git__Hash.{cmi,cmti} (exit 2)
# (cd _build/default && /home/opam/.opam/4.11/bin/ocamlc.opt -w -40 -g -bin-annot -I src/git/.git.objs/byte -I /home/opam/.opam/4.11/lib/angstrom -I /home/opam/.opam/4.11/lib/astring -I /home/opam/.opam/4.11/lib/bigarray-compat -I /home/opam/.opam/4.11/lib/bigarray-overlap -I /home/opam/.opam/4.11/lib/bigarray-overlap/stubs -I /home/opam/.opam/4.11/lib/bigstringaf -I /home/opam/.opam/4.11/lib/bytes -I /home/opam/.opam/4.11/lib/checkseum -I /home/opam/.opam/4.11/lib/cstruct -I /home/opam/.opam/4.11/lib/decompress -I /home/opam/.opam/4.11/lib/decompress/impl -I /home/opam/.opam/4.11/lib/digestif -I /home/opam/.opam/4.11/lib/duff -I /home/opam/.opam/4.11/lib/encore -I /home/opam/.opam/4.11/lib/eqaf -I /home/opam/.opam/4.11/lib/fmt -I /home/opam/.opam/4.11/lib/fpath -I /home/opam/.opam/4.11/lib/hex -I /home/opam/.opam/4.11/lib/ke -I /home/opam/.opam/4.11/lib/logs -I /home/opam/.opam/4.11/lib/lru -I /home/opam/.opam/4.11/lib/lwt -I /home/opam/.opam/4.11/lib/ocamlgraph -I /home/opam/.opam/4.11/lib/ocplib-endian -I /home/opam/.opam/4.11/lib/ocplib-endian/bigstring -I /home/opam/.opam/4.11/lib/optint -I /home/opam/.opam/4.11/lib/psq -I /home/opam/.opam/4.11/lib/result -I /home/opam/.opam/4.11/lib/rresult -I /home/opam/.opam/4.11/lib/seq -I /home/opam/.opam/4.11/lib/stdlib-shims -I /home/opam/.opam/4.11/lib/stringext -I /home/opam/.opam/4.11/lib/uri -no-alias-deps -open Git__ -o src/git/.git.objs/byte/git__Hash.cmi -c -intf src/git/hash.mli)
# File "src/git/hash.mli", line 18, characters 2-70:
# 18 |   S.HASH with type t = H.t and type ctx = H.ctx and type kind = H.kind
#        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
# Error: The signature constrained by `with' has no component named kind
```

**Comment by dinosaure** [comment]
Yes, it's expected :+1:, I will add constraints on Git.

File: packages/git/git.2.0.0/opam [hunk 1]
```diff
@@ -49,6 +49,9 @@ depends: [
   "tls"      {with-test}
   "mtime"    {with-test & >= "1.0.0"}
 ]
+conflicts: [
+  "digestif" {>= "1.0.0"}
```
Comment (mseri) [inline packages/git/git.2.0.0/opam:53]: Why not just add an upper bound `& < "1.0.0"` to the digestif dependency above instead?

Comment (mseri) [reply packages/git/git.2.0.0/opam:53]: Similarly for the other git packages

Comment (dinosaure) [reply packages/git/git.2.0.0/opam:53]: I got a time out from OPAM with such constraint :( I'm not sur why but it seems better to put the constraint as is.

Comment (mseri) [reply packages/git/git.2.0.0/opam:53]: Oh, weird. Let me ping @kit-ty-kate

Comment (mseri) [reply packages/git/git.2.0.0/opam:53]: Looks like it is still timing out in some cases though

Comment (hannesm) [reply packages/git/git.2.0.0/opam:53]: I'd consider `& < "1.0.0"` to be the more streamline solution. There are solver timeouts on various containers & old OCaml versions, or does it happen with a current (4.10+) system (e.g. your laptop)? If not, I'd be in favour of no conflicts but upper bounds instead.

Comment (dinosaure) [reply packages/git/git.2.0.0/opam:53]: I can retry with `& < "1.0.0"` but I'm worried about time-out :+1:.

**Comment by mseri** [comment]
The timeouts seems to be the more or less the same as with the other version, I think this is good to merge

**Pull request merged**